tests/specs/file.js

```diff
@@ -134,363 +195,576 @@ define(["dist/amd-resolver"], function(Resolver) {
+                    expect(parsedURI.protocolmark).to.equal("//");
+                  });
+
+                  it("hostname should equal `hoistedjs.com`", function() {
+                    expect(parsedURI.hostname).to.equal("hoistedjs.com");
+                  });
+
+                  it("port should be undefined", function() {
+                    expect(parsedURI.port).to.be.an('undefined');
+                  });
+
+                  it("path should equal `/moretesting/`", function() {
+                    expect(parsedURI.path).to.equal("/moretesting");
+                  });
+
+                  it("search should equal `?topic=data`", function() {
+                    expect(parsedURI.search).to.equal("?topic=data");
+                  });
+
+                  it("hash should equal `#migration/path`", function() {
+                    expect(parsedURI.hash).to.equal("#migration/path");
+                  });
+                });
+
+                describe("with a port", function() {
+                  describe("for `http://hoistedjs.com:599/moretesting?topic=data#migration/path`", function() {
+                    var parsedURI = File.parseUri("http://hoistedjs.com:599/moretesting?topic=data#migration/path");
+
+                    it("origin should equal `http://hoistedjs.com:599`", function() {
+                      expect(parsedURI.origin).to.equal("http://hoistedjs.com:599");
+                    });
+
+                    it("protocol should equal `http:`", function() {
+                      expect(parsedURI.protocol).to.equal("http:");
+                    });
+
+                    it("protocolmark should equal `//`", function() {
+                      expect(parsedURI.protocolmark).to.equal("//");
+                    });
+
+                    it("hostname should equal `hoistedjs.com`", function() {
+                      expect(parsedURI.hostname).to.equal("hoistedjs.com");
+                    });
+
+                    it("port should equal `599`", function() {
+                      expect(parsedURI.port).to.equal('599');
+                    });
+
+                    it("path should equal `/moretesting`", function() {
+                      expect(parsedURI.path).to.equal("/moretesting");
+                    });
+
+                    it("search should equal `?topic=data`", function() {
+                      expect(parsedURI.search).to.equal("?topic=data");
+                    });
+
+                    it("hash should equal `#migration/path`", function() {
+                      expect(parsedURI.hash).to.equal("#migration/path");
+                    });
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+
+
+      describe("using FILE protocol", function() {
+        it("with an empty string", function() {
+          var testFunction = File.parseUri.bind(File, "");
+          expect(testFunction).to.throw(Error, "Must provide a string to parse");
```

Comment:
The files tabs seems to be the better to add comment, I am doing it here. I didn't notice the `bind` in there until you mentioned it.  So is the expect calling `testFunction`?  That's an interesting way to do it...  Never really done it like that before.  I am really going to be introducing sinonjs to help us with that kind of tests.  But I think it would ok to leave it like that for now if that behaves as expected.